ckanext/restricted/controller.py

```diff
@@ -45,20 +48,26 @@ def __before__(self, action, **env):
             base.abort(401, _('Not authorized to see this page'))
 
     def _send_request_mail(self, data):
-        success = False
         try:
+            dataset_name = data['package_name']
+            resource_id = data['resource_id']
```

Comment:
This is so confusing 😂 
I'm not sure where you're getting `resource_id` from `data`

This is what my `data` object looks like:
```
{
    "resource": u"408988bf-3597-4989-96d0-84eaeccd88f1",
    u"pkg_dict": {
        u"num_resources": 1,
        u"maintainer": None,
        u"tags": [],
        u"relationships_as_object": [],
        u"private": False,
        u"maintainer_email": None,
        u"num_tags": 0,
        u"groups": [],
        u"license_id": None,
        u"isopen": False,
        u"id": u"c9051e79-c38b-4d0a-b026-a523ac778c70",
        u"metadata_created": u"2020-12-03T17:33:44.627910",
        u"organization": {
            u"description": u"Just another test organization.",
            u"created": u"2020-12-03T17:33:44.545848",
            u"title": u"Test Organization",
            u"name": u"test_org_01",
            u"is_organization": True,
            u"state": u"active",
            u"image_url": u"https://placekitten.com/g/200/100",
            u"type": u"organization",
            u"id": u"76fd462b-5dbd-49f5-ac40-e5e939c97703",
            u"approval_status": u"approved",
        },
        u"name": u"name",
        u"metadata_modified": u"2020-12-03T17:33:44.709981",
        u"author": None,
        u"author_email": None,
        u"notes": u"Just another test dataset.",
        u"license_title": None,
        u"owner_org": u"76fd462b-5dbd-49f5-ac40-e5e939c97703",
        u"relationships_as_subject": [],
        u"url": None,
        u"state": u"active",
        u"version": None,
        u"title": u"Test Dataset",
        u"creator_user_id": u"48b5710c-a4e3-45f3-8caf-20472da8f3dd",
        u"type": u"dataset",
        u"resources": [
            {
                u"hash": u"",
                u"description": u"Just another test resource.",
                u"cache_last_updated": None,
                u"package_id": u"c9051e79-c38b-4d0a-b026-a523ac778c70",
                u"mimetype_inner": None,
                u"url_type": None,
                u"id": u"408988bf-3597-4989-96d0-84eaeccd88f1",
                u"size": None,
                u"mimetype": None,
                u"cache_url": None,
                u"name": u"name",
                u"metadata_modified": u"2020-12-03T17:33:44.714185",
                u"created": u"2020-12-03T17:33:44.717580",
                u"url": u"http://link.to.some.data",
                u"restricted": '{"allowed_organizations": "", "allowed_users": "", "level": "public"}',
                u"format": u"res_format",
                u"state": u"active",
                u"last_modified": None,
                u"position": 0,
                u"resource_type": None,
            }
        ],
        u"extras": [],
    },
    "package_name": u"c9051e79-c38b-4d0a-b026-a523ac778c70",
    "maintainer_email": u"",
    "message": u"aaaa",
    "save": u"1",
}

```

You need to use `data['resource']` to get the `resource_id`

Can you confirm this and we can have a call and go over it again.
I've reset all my branches to the latest development with ckan 2.9.1 and am running this branch
Our local envs seem to be acting differently

Reply:
Conclusion: It's not that our dev envs were acting differently, it was down to tomek testing through the UI and me testing through unit tests with bugs in them. So it just comes down to fixing the tests which is what i've done now.